docs/PacketFence_MSPKI_Quick_Install_Guide.asciidoc

```diff
@@ -0,0 +1,313 @@
+
+Communication between MSPKI and PacketFence will be using port 80 or 443 depending on your configuration.
+
+Configuring Network Device Enrollement Service (NDES)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Our main activity in the deployment of ADCS is to configure Network Enrollement Device Service (NDES) which will allow us to exchange certificate with the MS PKI server via the protocol Simple Certificate Exchange Protocol (SCEP).
+
+Every configuration change has to be done by an account with administrative privileges.
+
+To allow the use of NDES as a Pki-Provider in BOYD deployement there is multiple changes.
+
+Challenge Password
+^^^^^^^^^^^^^^^^^^
+
+MSCEP include by default a challenge password, which is unique and dynamically generated for each device which want to enroll on the system. The issue here while using the challenge password is that it is necessary for an administrator to connect to the NDES server from the user's device. This a barrier to BOYD deployement for instance, where you want the user to be able to register himself his device. Since we use NDES with PacketFence, our security to obtain a certificate would be the credetials necessary to access the enrollement system.
+
+To disable the challenge password you need to modifiy the following key in the Windows registry. 
+
+Click `Start` and enter `regedit`.
+
+Navigate to `Computer` > `HKEY_LOCAL_MACHINE` > `SOFTWARE` > `Microsoft` > `Cryptography` > `MSCEP` > `EnforcePassword`.
+
+Change the value of `EnforcePassword` to `0` (default is `1`).
+
+Extend URL length for the request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Best practices from Cisco recommand to extend the URL length to avoid issue with longer request. 
+
+To do so, enter the following command in the CLI on the NDES server:
+
+ %systemroot%\system32\inetsrv\appcmd.exe set config /section:system.webServer/
+ security/requestFiltering /requestLimits.maxQueryString:"8192" /commit:apphost
+
+Certificate template 
+^^^^^^^^^^^^^^^^^^^^
+
+NOTE: Remember that the validity of your CA can impact your whole certificate architecture.
+
+The main goal is to deliver certificate for either `user Authentication` or `machine Authentication`, this mean we have to setup a specific template.
+
+First, indepandent of which type of authentication you will use, the certificate template need at least the following `Enhanced Key Usage`:
+
+* Client Authentication
+
+and the following `Key Usage`:
+
+* Digital signature
+* Key encipherment
+
+Then if you want to use machineAuth you should duplicate the template `Computer` and change the settings that suits your use of certificate. To force a template to deliver Machine certificates, go to the tab `Subject Name` and select `Type of subject` > `Computer or other device`.
```

Comment:
Subject Name` and select `Type of subject` > `Computer or other device` ? where did you see it ?